src/test/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitterTest.kt

```diff
@@ -290,14 +292,14 @@ internal class HMPPSDomainEventsEmitterTest {
             "MDI"
           )
         )
-      emitter!!.convertAndSendWhenSignificant(
+      emitter.convertAndSendWhenSignificant(
         OffenderEvent.builder()
           .eventType("OFFENDER_MOVEMENT-RECEPTION")
           .offenderIdDisplay("A1234GH")
           .eventDatetime(LocalDateTime.parse("2020-12-04T10:42:43"))
           .build()
       )
-      Mockito.verifyNoInteractions(awsHMPPSEventsSnsClient)
+      Mockito.verify(hmppsEventSnsClient, atLeastOnce()).publishAsync(publishRequestCaptor!!.capture())
```

Comment:
Looking at a better solution to atLeastOnce() - seems like a bit of a hack

Reply:
Why do you need atLastOnce?